cheroot/server.py

```diff
@@ -1255,30 +1260,26 @@ def __init__(self, server, sock, makefile=MakeFile):
         )
 
     def communicate(self):
-        """Read each request and respond appropriately."""
+        """Read each request and respond appropriately.
+
+        Returns true if the connection should be kept open.
+        """
         request_seen = False
         try:
-            while True:
-                # (re)set req to None so that if something goes wrong in
-                # the RequestHandlerClass constructor, the error doesn't
-                # get written to the previous request.
-                req = None
-                req = self.RequestHandlerClass(self.server, self)
-
-                # This order of operations should guarantee correct pipelining.
-                req.parse_request()
-                if self.server.stats['Enabled']:
-                    self.requests_seen += 1
-                if not req.ready:
-                    # Something went wrong in the parsing (and the server has
-                    # probably already made a simple_response). Return and
-                    # let the conn close.
-                    return
+            req = self.RequestHandlerClass(self.server, self)
+            req.parse_request()
+            if self.server.stats['Enabled']:
+                self.requests_seen += 1
+            if not req.ready:
+                # Something went wrong in the parsing (and the server has
+                # probably already made a simple_response). Return and
+                # let the conn close.
+                return
```

Comment:
I don't recall if I asked this before but it feels like this should return `False` explicitly. WDYT?

Reply:
I agree, I've made changes for it.